wdl-format/src/lib.rs

```diff
@@ -0,0 +1,191 @@
+//! A library for auto-formatting WDL code.
+
+#![warn(missing_docs)]
+#![warn(rust_2018_idioms)]
+#![warn(rust_2021_compatibility)]
+#![warn(missing_debug_implementations)]
+#![warn(clippy::missing_docs_in_private_items)]
+#![warn(rustdoc::broken_intra_doc_links)]
+
+use anyhow::Result;
+use wdl_ast::AstNode;
+use wdl_ast::AstToken;
+use wdl_ast::Comment;
+use wdl_ast::Diagnostic;
+use wdl_ast::Direction;
+use wdl_ast::Document;
+use wdl_ast::Ident;
+use wdl_ast::SyntaxElement;
+use wdl_ast::SyntaxKind;
+use wdl_ast::SyntaxNode;
+use wdl_ast::Validator;
+use wdl_ast::Version;
+use wdl_ast::VersionStatement;
+
+mod comments;
+mod import;
+mod metadata;
+mod state;
+mod task;
+mod v1;
+mod workflow;
+
+use comments::format_inline_comment;
+use comments::format_preceding_comments;
+use state::State;
+
+/// Newline constant used for formatting.
+pub const NEWLINE: &str = "\n";
```

Comment:
It sounded to me like we were settled on the addition of a `wdl-config` crate? My thinking is that hardcoded *nix newlines is fine, with the understanding that a "proper" implementation is incoming. So my opposition here is to avoid duplicating efforts.

Reply:
I think adding a `#[cfg(windows)]` to default it to `\r\n` makes sense without configuration and then adding a configuration that overrides it in the future (i.e. `newline = "unix"` or `newline = "windows"`), but we could have a constant for the default new line and that should be platform specific. Plus, it would get the expected output on Windows in the interim until further configuration comes along.

Reply:
Fair enough, I'll add that!